.github/workflows/test.yml

```diff
@@ -0,0 +1,52 @@
+# if changing this name, also update promotion.yaml
+name: test
+
+on:
+  push:
+    branches-ignore:
+      - master
```

Comment:
hmm, can't we use `if: github.ref == 'refs/heads/master'` -> make push canary in this file instead to save duplication?

Reply:
But I would like to avoid loading the secrets in a workflow that runs on every branch.

I am not sure what the implementation is, not sure if we can prevent loading the secrets with `if`.

Reply:
Is there a way you can stop the secrets being loaded in a non-master branch anyway?

Reply:
Ahh I see, you can use environments...